core/array/bsearch_index_spec.rb

```diff
@@ -0,0 +1,97 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require File.expand_path('../../enumerable/shared/enumeratorized', __FILE__)
+
+ruby_version_is "2.3" do
+  describe "Array#bsearch_index" do
+    context "when not passed a block" do
+      before :each do
+        @enum = [1, 2, 42, 100, 666].bsearch_index
+      end
+
+      it "returns an Enumerator" do
+        @enum.should be_an_instance_of(enumerator_class)
+      end
+
+      it "returns nil for #size" do
+        @enum.size.should be_nil
+      end
+
+      it "returns index of element given condition in block" do
+        @enum.each { |x| x >= 33 }.should == 2
+      end
+    end
+
+    it "raises TypeError when block passed wrong argument" do
+      lambda { [1, 2, 3].bsearch_index { "not ok" } }.should raise_error(TypeError)
+    end
+
+    it "returns nil when passed empty block" do
+      [1, 2, 3].bsearch_index {}.should be_nil
+    end
+
+    it "returns nil when passed block with false" do
+      [1, 2, 3].bsearch_index { false }.should be_nil
+    end
+
+    context "minimum mode" do
+      before :each do
+        @array = [0, 4, 7, 10, 12]
+      end
+
+      it "x >= 4" do
```

Comment:
RubySpec examples should have a description of the behavior, so in this case it should describe the mode like
`returns the index of the first element for which the block yields true`
And then the three first calls could be grouped into that.
The last one should be in its own example saying
`returns nil if there is no element for which the block yields true`